JavaScript/JavaScriptSDK/AppInsights.ts

```diff
@@ -407,10 +407,19 @@ module Microsoft.ApplicationInsights {
 
         /**
          * Immediately send all queued telemetry.
+         * @param {boolean} async - If flush should be call asynchronously
          */
-        public flush() {
+        public flush(async?: boolean) {
```

Comment:
Can you use `public flush(async = true)`? Then you don't have to check if async is boolean. 

The same improvement can be done in the [sender.triggerSend()](https://github.com/Microsoft/ApplicationInsights-JS/blob/2745ffadaa834ad7532182e5aa4417b1cb7907c8/JavaScript/JavaScriptSDK/Sender.ts#L256) method. 

Thanks!